knowledge_repo/app/index.py

```diff
@@ -67,23 +67,23 @@ def update_index_required():
         db_session.commit()
 
 
-def update_index():
+def update_index(force=False):
     """
     Initialize the db from a KnowledgeRepository object
     """
 
-    if not update_index_required():
+    if not force and not update_index_required():
```

Comment:
@danfrankj Perhaps we should be a little careful if an index is already underway. At the moment we have no way of cancelling an index, and so I think we should avoid starting another index if one is already happening.

`if not force and not update_index_required() or is_indexing():`